File: src/main/LuaFiles/StationRules.lua
--
-- Created by IntelliJ IDEA.
-- User: userhp
-- Date: 23/02/2016
-- Time: 11:30
-- To change this template use File | Settings | File Templates.
--
local oneStationRent = 25
local twoStationRent = 50
local threeStationRent = 100
local fourStationRent = 200

function calculateRent(owner, visitor, groupStation, moveTypeCard)
    rentOwed = 0
    stationsOwned = owner:ownsSpacesOfGroup(groupStation)

    if stationsOwned == 4 then rentOwed = fourStationRent

    elseif stationsOwned == 3 then rentOwed = threeStationRent

    elseif stationsOwned == 2 then rentOwed = twoStationRent

    else rentOwed = oneStationRent
    end -- end if

    if visitor:getMoveTaken():equals(moveTypeCard) then rentOwed = rentOwed * 2
    end

    return rentOwed
end

File: src/main/java/Rules/StationRules.java
package Rules;

import Board.Group;
import Players.Player;

/**
 * Created by Lucy on 2018/04/02.
 */
public class StationRules {
    private static int oneStationRent;
    private static int twoStationRent;
    private static int threeStationRent;
    private static int fourStationRent;

    public StationRules() {
        oneStationRent = 25;
        twoStationRent = 50;
        threeStationRent = 100;
        fourStationRent = 200;
    }
    public int calculateRent(Player owner, Player visitor) {
        int rentOwed = 0;
        int stationsOwned = owner.ownsSpacesOfGroup(Group.Station);
        if (stationsOwned == 4) {
            rentOwed = fourStationRent;
        } else if (stationsOwned == 3) {
            rentOwed = threeStationRent;
        } else if (stationsOwned == 2) {
            rentOwed = twoStationRent;
        } else {
            rentOwed = oneStationRent;
        }
        if (visitor.getMoveTaken().equals(MoveType.Card)) {
            rentOwed = rentOwed * 2;
        }
        return rentOwed;
    }
}
